File: MCC/2023_08_11/A/5.cpp
#include <bits/stdc++.h>
#include <atcoder/all>
using namespace std;
using namespace atcoder;

#define all(v) v.begin(), v.end()
#define sort(v) sort(v.begin(), v.end())
#define reverse(v) reverse(v.begin(), v.end())
#define ll int64_t
#define ld long double

ld A_log(ll shinsuu, ll tei)
{
    return log(shinsuu) / log(tei);
}

int main()
{
    ll X, Y, A, B;
    cin >> X >> Y >> A >> B;
    ll ans = 0;
    // いつX*(A^n)がX+B*nを超えるかのnをにぶたんで求める
    ll left = -1;
    ll right = 1e18 + 10;
    while (abs(right - left) > 1)
    {
        ll mid = (left + right) / 2;
        // if (X * pow(A, mid) >= X + B * mid)
        if (A_log(X, A) + mid - A_log(mid, A) >= A_log(X / (ld)mid + B, A) || A_log(X, A) + mid >= A_log(Y, A))
        {
            right = mid;
        }
        else
        {
            left = mid;
        }
    }
    ll n = left;
    ans += n;

    ll left2 = -1;
    ll right2 = 1e18 + 10;
    while (abs(right2 - left2) > 1)
    {
        ll mid = (left2 + right2) / 2;
        // if (X * pow(A, n) + B * mid >= Y)
        // if (X * pow(A, n) / mid >= Y / mid - B)
        if (Y / (ld)mid - B < 0 || X < 0 || A_log(X, A) + n - A_log(mid, A) >= A_log(Y / (ld)mid - B, A))
        {
            right2 = mid;
        }
        else
        {
            left2 = mid;
        }
    }
    ans += left2;
    cout << ans << endl;
}

File: MCC/2023_08_11/A/5.py
# #include <bits/stdc++.h>
# #include <atcoder/all>
# using namespace std;
# using namespace atcoder;

# #define all(v) v.begin(), v.end()
# #define sort(v) sort(v.begin(), v.end())
# #define reverse(v) reverse(v.begin(), v.end())
# #define ll int64_t
# #define ld long double

# int main()
# {
#     ll X, Y, A, B;
#     cin >> X >> Y >> A >> B;
#     // *A < +BまでAを使う
#     ll ans = 0;
#     ll sum = X;
#     while (X * A < X + B)
#     {
#         X *= A;
#         ans++;
#         if (X >= Y)
#         {
#             cout << ans - 1 << endl;
#             return 0;
#         }
#     }
#     // *A < +BならBを使う
#     while (true)
#     {
#         X += B;
#         ans++;
#         if (X >= Y)
#         {
#             cout << ans - 1 << endl;
#             return 0;
#         }
#     }
# }

# C++プログラムをPythonに変換

X, Y, A, B = map(int, input().split())
ans = 0
while X * A < X + B:
    X *= A
    ans += 1
    if X >= Y:
        print(ans - 1)
        exit()
while True:
    X += B
    ans += 1
    if X >= Y:
        print(ans - 1)
        exit()
